pkg/asset/installconfig/gcp/dns.go

```diff
@@ -29,6 +31,21 @@ func GetPublicZone(ctx context.Context, project, baseDomain string) (*dns.Manage
 	return dnsZone, nil
 }
 
+// ParseZoneName will attempt to extract the project and zone from a long zone name that
+// includes the full path (i.e. projects/{project-id}/managedZones/{managed-zone-name}). If the data
+// cannot be parsed or does not include this information, the default (provided) values will be returned
+func ParseZoneName(providedZoneName string) (string, string, error) {
+
+	switch splitZoneName := strings.Split(providedZoneName, "/"); {
+	case len(splitZoneName) == 1:
+		return "", providedZoneName, nil
+	case len(splitZoneName) >= 4 && splitZoneName[0] == "projects" && splitZoneName[2] == "managedZones":
```

Comment:
In this case, len(splitZoneName) can only be 4.... anything else would have gibberish on the end, and could be considered invalid.

Reply:
It could be 3 couldn't it. We don't force this check in a specific sequence. What if someone entered something bad here like `projects/<project-id>/managedZones`. I hope they don't do this, but if they do then the splitZoneName[3] would cause an index out of bounds. 

Or am I just overthinking and adding too much error handling?

Reply:
Right, 3 would be too little... because there needs to be 4 segments... (where splitZoneName[3] is the 4th one).

Reply:
```
case len(splitZoneName) == 4 && splitZoneName[0] == "projects" && splitZoneName[2] == "managedZones":
```